compiler/src/main/java/org/qbicc/type/definition/classfile/ClassFileImpl.java

```diff
@@ -324,7 +325,13 @@ public Literal getConstantValue(int idx, TypeParameterContext paramCtxt) {
         }
         int constantType = getConstantType(idx);
         switch (constantType) {
-            case CONSTANT_Class: return setIfNull(literals, idx, literalFactory.literalOfType(getTypeConstant(idx, paramCtxt)));
+            case CONSTANT_Class: {
+                ValueType valueType = getTypeConstant(idx, paramCtxt);
```

Comment:
I think we want to forbid `getTypeConstant` from returning `ReferenceType`.  Instead, in places that require a reference type, it's easier to produce a reference type from an object type than the reverse.

Reply:
I am realizing it now what you meant when you said it is a bit nontrivial ;-)